### PR TITLE
fix: add cert redirect for Netlify

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Netlify does not automatically discover client only routes.  This PR creates a redirect telling it where to find the HTML for certifications.

Related to #37378, but only fixes the style issues that appear when the page is reloaded.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
@raisedadead I've tested this locally and, once you create the `_redirects` file, `netlify dev` seems to work.

I'm not sure if the other client-only routes need looking at (presumably they do).

**Comment by raisedadead** [comment]
You are correct, this will fix only direct call to the client app externally. This change is also a URL rewrite, that Netlify will fetch the page behind the scene keeping the URL same.

Can you ditch the /index.html and see it still has the same effect?

Thanks.

**Comment by ojeytonwilliams** [comment]
Sure.  I couldn't see a difference, so here's the concise version.

File: tools/scripts/build/__snapshots__/create-redirects.test.js.snap [hunk 1]
```diff
@@ -48,6 +48,7 @@ https://freecodecamp-org.netlify.com/*        https://www.freecodecamp.org/:spla
 /:username/data-visualization-certification   /certification/:username/legacy-data-visualization 301
 /:username/back-end-certification             /certification/:username/legacy-back-end 301
 /:username/full-stack-certification           /certification/:username/full-stack 301
+/certification/*                              /certification 200
```
Comment (raisedadead) [inline tools/scripts/build/__snapshots__/create-redirects.test.js.snap]:
```suggestion
/certification/*                              /certification/:splat 200
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
My sincerest apologies @ojeytonwilliams, I just woke up an hour ago. I need my coffee. 

Without these (fresh change suggested) you will not be able to reach anyone's cert from outside, because it would redirect to `/certification` which will take you to `/`

**Comment by ojeytonwilliams** [comment]
@raisedadead no worries, I wrote this during a bout of insomnia.  Should have just waited until I was thinking clearly.

Anyway, I tried the `:splat` redirect, but that doesn't seem to work.  Here's what I get:

CertWithNetlify

No 404 this time, but the `light-palette` class is still getting applied.  The network log for reload is:

NetworkOutput

Steps I took:

1. put `FREECODECAMP_NODE_ENV='production'` in `.env`
2. `npm run build:client`
3. remove the environment variable (I'm missing some keys) and `npm run develop:server` 
4. `cd client/public`
5. `netlify dev`
6. navigate to a cert and reload

What I *think* is happening, is that by adding `:splat` we're telling Netlify to actually look in the directories `certification/username/certName`, but there isn't anything there.  Whereas, without `:splat` it serves up whatever's in `certification/` which is a single `index.html` page which is expecting to be given the parameters `username` and `certName` by reach.  And somehow that actually happens.

Or I've done something dumb and this won't work in production.  Any ideas?

**Comment by raisedadead** [comment]
Yup, you are correct. 

This is a chicken and egg problem, and the whole thing needs a rejig.

What keys are you missing, I can give you access to all those, hit me on Signal?

**Pull request closed without merging**